docs/dependency-injection/dependency-injection.md

```diff
@@ -0,0 +1,220 @@
+
+Here's an example describing how to use the app orchestrator to provide services as part of a feature:
+
+```ts
+export const app = appBuilder()
+  .withFeature(someFeature)
+  // ...
+  .withFeature({
+    providers: [{ provide: CartService, useClass: CustomizedCartService }],
+  })
+  // ...
+  .create();
+````
+
+In this example, customized cart service is provided by specifying the providers property within a feature object. This way, you can easily manage and extend the services provided by your Oryx application.
+
+# Injecting INJECTOR
+
+In some cases, dependencies may be optional or resolved at runtime, making it difficult to inject them directly into a service. In these situations, you can inject the current DI container using the `INJECTOR` token, like in the following example:
+
+
+```ts
+constructor(protected injector = inject(INJECTOR)) {}
+```
+
+
+With the `INJECTOR` token injected, you can use the injector's `inject` method to dynamically resolve dependencies at runtime. For example:
+
+```ts
+someServiceMethod(field) {
+	const template = this.injector.inject(`${FormFieldRenderer}-${field.type}`);
+}
+
+```
+
+In this example, the `inject` method is used to resolve a dependency based on a dynamic token that includes the field type. Since the field type is only known at runtime, it is not possible to inject the dependency directly. Instead, the `INJECTOR` token is injected, allowing the `inject` method to resolve the dependency dynamically.
+
+
+
+# Multi Providers
+
+While most dependencies in an app correspond to only one value, such as a class, there are situations where we want to have dependencies with multiple different values, for example HTTP interceptors or normalizers. However, it's not very practical to configure all these dependencies separately, because we want to access them all together at once. Therefore, we can use a special type of dependency that accepts multiple values, not just one, linked to the exact same dependency injection token. These are called multi-providers.
+
+## Providing
+
+Providers which token naming is ended with asterisk (`*`) `[token-base-name]*`:
+
+- creates multi providers with array of elements.
+
+```ts  
+[  
+  {    provider: 'multi*',    useValue: 'a',  },  {    provider: 'multi*',    useValue: 'b',  },  {    provider: 'multi*',    useValue: 'c',  },];  
+///  
+inject('multi*'); // [a,b,c];  
+```  
+
+Providers which token naming is included with asterisk (`*`) `[token-base-name]*[token-specifier-name]`:
+
+- creates single provider by token `[token-base-name]*[token-specifier-name]`.
+- creates multi providers with array of elements by token `[token-base-name]*`.
+
+```ts  
+[  
+  {    provider: 'multi*a',    useValue: 'a',  },  {    provider: 'multi*b',    useValue: 'b',  },  {    provider: 'multi*c',    useValue: 'c',  },];  
+///  
+inject('multi*a'); // a;  
+inject('multi*b'); // b;  
+inject('multi*c'); // c;  
+inject('multi*'); // [a,b,c];  
```

Comment:
Isn't it just a duplication of previous example?

Reply:
simplified